integration/integration_test.go

```diff
@@ -75,9 +80,41 @@ func runCommands(cmds []*command, t *testing.T) {
 	}
 }
 
+func cleanupFunc(cleanupDone chan bool, cleanupCommands []*command) {
+	if cancelling {
+		// Wait until the other call of cleanupFunc() is done.
+		<-cleanupDone
+		return
+	}
+
+	cancelling = true
```

Comment:
I don't get why you need a channel here. If I understand correctly the behavior should be to avoid running cleanupFunc multiple times. Isn't enough to use the cancelling variable? (the atomic version of it)?

Reply:
For instance, if the workflow run is cancelled during the `initCommands`, the `cleanupFunc()` will be called twice: Once from the [defer](https://github.com/kinvolk/inspektor-gadget/blob/f847eb2daa1773392ff7897ce8f550dae9ee1239/integration/integration_test.go#L237) and the other from [the signal handler](https://github.com/kinvolk/inspektor-gadget/blob/f847eb2daa1773392ff7897ce8f550dae9ee1239/integration/integration_test.go#L214). At this point, `cancelling` will avoid running `cleanupFunc()` multiple times but if we let one of them finish, it will make the main goroutine to finish without waiting for the `cleanupFunc()`  to finish. And that's where the `cleanupDone` channel helps. Does it make sense for you?

Reply:
As agreed, I kept that channel but moved it outside the `cleanupFunc()` as it can be managed outside and makes the code clearer.